src/execution/__tests__/oneof-test.ts

```diff
@@ -181,5 +186,338 @@ describe('Execute: Handles OneOf Input Objects', () => {
+          }
+        }
+      `;
+      const result = executeQuery(query, rootValue, { a: undefined });
+
+      expectJSON(result).toDeepEqual({
+        data: {
+          test: null,
+        },
+        errors: [
+          {
+            // A nullable variable in a oneOf field position would be caught at validation-time
+            // hence the vague error message here.
+            message:
+              'Argument "input" of type "TestInputObject!" has invalid value { a: $a }.',
+            locations: [{ line: 3, column: 23 }],
+            path: ['test'],
+          },
+        ],
+      });
+    });
+
+    it('rejects nulled variable for field', () => {
+      const query = `
+        query ($a: String) {
+          test(input: { a: $a }) {
+            a
+            b
+          }
+        }
+      `;
+      const result = executeQuery(query, rootValue, { a: null });
+
+      expectJSON(result).toDeepEqual({
+        data: {
+          test: null,
+        },
+        errors: [
+          {
+            // A nullable variable in a oneOf field position would be caught at validation-time
+            // hence the vague error message here.
+            message:
+              'Argument "input" of type "TestInputObject!" has invalid value { a: $a }.',
+            locations: [{ line: 3, column: 23 }],
+            path: ['test'],
+          },
+        ],
+      });
+    });
+
+    it('rejects missing variable for field', () => {
+      const query = `
+        query ($a: String) {
+          test(input: { a: $a }) {
+            a
+            b
+          }
+        }
+      `;
+      const result = executeQuery(query, rootValue);
+
+      expectJSON(result).toDeepEqual({
+        data: {
+          test: null,
+        },
+        errors: [
+          {
+            // A nullable variable in a oneOf field position would be caught at validation-time
+            // hence the vague error message here.
+            message:
+              'Argument "input" of type "TestInputObject!" has invalid value { a: $a }.',
+            locations: [{ line: 3, column: 23 }],
+            path: ['test'],
+          },
+        ],
+      });
+    });
+
+    it('rejects missing second variable for field', () => {
+      const query = `
+        query ($a: String, $b: String) {
+          test(input: { a: $a, b: $b }) {
+            a
+            b
+          }
+        }
+      `;
+      const result = executeQuery(query, rootValue, { a: '123' });
+
+      expectJSON(result).toDeepEqual({
+        data: {
+          test: null,
+        },
+        errors: [
+          {
+            // A nullable variable in a oneOf field position would be caught at validation-time
+            // hence the vague error message here.
+            message:
+              'Argument "input" of type "TestInputObject!" has invalid value { a: $a, b: $b }.',
+            locations: [{ line: 3, column: 23 }],
+            path: ['test'],
+          },
+        ],
```

Comment:
I had to validate that this is the correct behavior (e.g. do we care about the uncoerced value during coercion) - but yes, we do. Good catch :+1: 

> - If the input object literal or **unordered map** does not contain exactly one entry an error must be thrown.
> - Within the input object literal or unordered map, if the single entry is {null} an error must be thrown.
> - If the **coerced unordered map** does not contain exactly one entry an error must be thrown.
> - If the value of the single entry in the coerced unordered map is {null} an error must be thrown.